scaleway/resource_vpc_public_gateway_ip.go

```diff
@@ -153,10 +154,19 @@ func resourceScalewayVPCPublicGatewayIPDelete(ctx context.Context, d *schema.Res
 		return diag.FromErr(err)
 	}
 
-	err = vpcgwAPI.DeleteIP(&vpcgw.DeleteIPRequest{
-		IPID: ID,
-		Zone: zone,
-	}, scw.WithContext(ctx))
+	err = retryVPCContext(ctx, retryGWTimeout, func() *resource.RetryError {
+		errDeleteIP := vpcgwAPI.DeleteIP(&vpcgw.DeleteIPRequest{
+			IPID: ID,
+			Zone: zone,
+		}, scw.WithContext(ctx))
+		if errDeleteIP != nil {
+			if is409Error(errDeleteIP) || is412Error(errDeleteIP) {
+				return resource.RetryableError(errDeleteIP)
```

Comment:
Can this go into an infinite loop?

Reply:
Yes, It is recommended in this case. This resource could been used by another resource and until today we don't have a way to verify before deleted